pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -62,6 +63,7 @@ const (
 	errStoreProvider         = "could not get store provider"
 	errStoreClient           = "could not get provider client"
 	errGetExistingSecret     = "could not get existing secret: %w"
+	errClusterStoreMismatch  = "cluster store %s does not match namespace %s"
```

Comment:
small nit: i think we can improve the error message by stating the specific reason why it was denied instead of just saying it didn't match:
```
cluster store fake does not match namespace foo
```

```suggestion
	errClusterStoreMismatch  = "using cluster store %q is not allowed from namespace %q: denied by spec.condition"
```

That should give us:
```
using cluster store "fake" is not allowed from namespace "foo": denied by spec.condition
```
Which i find more clear. What do you think?

Reply:
Sounds much better, added :)